h5py/_hl/group.py

```diff
@@ -217,17 +231,21 @@ def create_dataset_like(self, name, other, **kwupdate):
         if other.maxshape != other.shape:
             kwupdate.setdefault('maxshape', other.maxshape)
 
-        return self.create_dataset(name, **kwupdate)
+        return self.create_dataset(name, dcpl=other._dcpl, **kwupdate)
```

Comment:
This is a nice payoff of letting dcpl go it to this method, but I still lean to not exposing it....